amplify/backend/function/fetchMotionState/src/index.js

```diff
@@ -34,8 +37,73 @@ exports.handler = async (event) => {
           motionData,
         );
       }
+
+      if (!motionStateHistory.hasFailedNotFinalizable) {
+        await updateMotionMessagesInDB(
+          transactionHash,
+          motionData,
+          ['MotionHasFailedNotFinalizable'],
+          'hasFailedNotFinalizable',
+        );
+      }
+    }
+
+    const didPass = didMotionPass(motionData);
+
+    if (
+      motionState === MotionState.Reveal &&
+      !motionStateHistory.inRevealPhase
+    ) {
+      await updateMotionMessagesInDB(
+        transactionHash,
+        motionData,
+        ['MotionRevealPhase'],
+        'inRevealPhase',
+      );
+    }
+
+    if (
+      motionState === MotionState.Finalizable ||
+      motionState === MotionState.Finalized
+    ) {
+      // Check if the motion passed and the messages have not already been stored in the db
+      if (didPass && !motionStateHistory.hasPassed) {
+        const newMessages = [];
+
+        // only display voting results if a vote has occurred
+        if (motionStateHistory.hasVoted) {
+          newMessages.push('MotionRevealResultMotionWon');
+        }
+        newMessages.push('MotionHasPassed');
+
+        await updateMotionMessagesInDB(
+          transactionHash,
+          motionData,
+          newMessages,
+          'hasPassed',
+        );
+      }
+
+      if (!didPass && !motionStateHistory.hasFailed) {
+        const newMessages = [];
+
+        // only display voting results if a vote has occurred
+        if (motionStateHistory.hasVoted) {
+          newMessages.push('MotionRevealResultObjectionWon');
+        }
+        newMessages.push('MotionHasFailedFinalizable');
+
+        await updateMotionMessagesInDB(
+          transactionHash,
+          motionData,
+          newMessages,
+          'hasFailed',
+        );
+      }
     }
+
+    return motionState;
   }
 
-  return motionState;
+  return undefined;
```

Comment:
This lamda is typed to return a number so let's return MotionState.Null as the fallback